src/audio/component.c

```diff
@@ -142,7 +142,8 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 		break;
 	case COMP_TRIGGER_STOP:
 	case COMP_TRIGGER_XRUN:
-		if (dev->state == COMP_STATE_ACTIVE) {
+		if (dev->state == COMP_STATE_ACTIVE ||
+		    dev->state == COMP_STATE_PREPARE) {
```

Comment:
WTH?
If you are in PREPARE state you set the component to PREPARE? I call this a no-op...

Reply:
I will think about your question.
Actually it works! I tested it on my GP, UP^2, BYT.